src/core/logger.cpp

```diff
@@ -15,7 +15,7 @@ namespace // anonymous
 {
 
 // Helper to throw from not implemented logger_impl methods.
-void throw_not_supported()
+[[noreturn]] void throw_not_supported()
```

Comment:
This should be in its own PR, ideally.

Reply:
This was necessary to get things to compile